internal/provider/resource_vdb_test.go

```diff
@@ -58,6 +78,79 @@ func testAccCheckDctVDBConfigBasic() string {
 	`, datasource_id)
 }
 
+func testAccCheckDctVDBBookmarkConfigBasic() string {
+	var bookmark_id, vdb_id string
+
+	// init client
+	cfg := dctapi.NewConfiguration()
+	cfg.Host = os.Getenv("DCT_HOST")
+	cfg.Scheme = "https"
+	cfg.HTTPClient = &http.Client{Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}}
+	cfg.AddDefaultHeader("Authorization", "apk "+os.Getenv("DCT_KEY"))
+	client := dctapi.NewAPIClient(cfg)
+
+	// create vdb
+	provisionVDBBySnapshotParameters := dctapi.NewProvisionVDBBySnapshotParameters()
+	provisionVDBBySnapshotParameters.SetAutoSelectRepository(true)
+	provisionVDBBySnapshotParameters.SetSourceDataId(os.Getenv("DATASOURCE_ID"))
+
+	vdb_req := client.VDBsApi.ProvisionVdbBySnapshot(context.Background())
+
+	vdb_res, vdb_http_res, vdb_err := vdb_req.ProvisionVDBBySnapshotParameters(*provisionVDBBySnapshotParameters).Execute()
+	if diags := apiErrorResponseHelper(vdb_res, vdb_http_res, vdb_err); diags != nil {
+		println("An error occured during vdb creation: " + vdb_err.Error())
+		return "" // return empty config to indicate config error
+	}
+	vdb_id = *vdb_res.VdbId
```

Comment:
What's nice is that once we'll support creating the bookmark from terraform, this will become something like:
```
resource "delphix_vdb" vdb1 {
   provision_type = "snapshot"
   auto_select_repository = true
   data_source_id = "%s"
}
resource "delphix_bookmark" bm {
   name = "my-bookmark"
   vdb_ids = [delphix_vdb.vdb1.id]
}

resource "delphix_vdb" "vdb_bookmark" {
  provision_type         = "my-bookmark"
  auto_select_repository = true
  bookmark_id            = blah
}
```